pages/showcase.tsx

```diff
@@ -0,0 +1,296 @@
+import { useState, useCallback, useMemo } from 'react'
+import {
+  Grid,
+  GridItem,
+  Box,
+  Flex,
+  Heading,
+  Text,
+  Link,
+  HStack,
+} from '@chakra-ui/layout'
+import type { ResponsiveValue } from '@chakra-ui/react'
+import { TabList, Tabs, Tab, TabPanels, TabPanel } from '@chakra-ui/tabs'
+import { SkipNavContent, SkipNavLink } from '@chakra-ui/skip-nav'
+import { useColorModeValue } from '@chakra-ui/color-mode'
+import { LinkIcon, Icon } from '@chakra-ui/icons'
+import SEO from 'components/seo'
+import Header from 'components/header'
+import ChakraNextImage from 'components/chakra-next-image'
+import DiscordStrip from 'components/discord-strip'
+import { AdBanner } from 'components/chakra-pro/ad-banner'
+import Footer from 'components/footer'
+import { t } from 'utils/i18n'
+import _ from 'lodash'
+import showcaseData from '../configs/showcase.json'
+import type {
+  IShowcase,
+  ShowcaseItem,
+} from '../scripts/showcase-preview-images'
+
+const categories = Object.keys(showcaseData as IShowcase)
+const categoriesWithAll = ['all', ...categories]
+
+const allItems: ShowcaseItem[] = categories.reduce((acc, cur) => {
+  return [...acc, ...showcaseData[cur]]
+}, [])
+
+const Showcase = () => {
+  const [index, setIndex] = useState<number>(0)
+
+  const handleTabsChange = useCallback((index: number) => {
+    if (index < 0 || index >= categories.length) setIndex(0)
+    setIndex(index)
+  }, [])
+
+  /*
+    Set hash in the first render
+
+    Troubleshooting
+    If using hash to get key, it might happen Error: Cancel rendering route
+    More Info: https://github.com/vercel/next.js/issues/2476
+   */
+  // useEffect(() => {
+  //   const hash = window?.location?.hash
+  //   if (hash) {
+  //     const hashStringWithoutSign = removeHashSign(hash)
+  //     const categoryIndex = getHashIndexInCategory(hashStringWithoutSign)
+  //     setIndex(categoryIndex !== -1 ? categoryIndex : 0)
+  //   }
+  // }, [])
```

Comment:
If you don't use this you can remove it. Cleans up the code.

Reply:
Sure! Thanks for pointing it out!